xwiki-rendering-transformations/xwiki-rendering-transformation-macro/src/main/java/org/xwiki/rendering/macro/descriptor/ContentDescriptor.java

```diff
@@ -36,4 +43,16 @@
      * @return indicate if the macro content is mandatory.
      */
     boolean isMandatory();
+
+    /**
+     * This method will return the type of the macro content. By default it fallback to List&lt;Block&gt;.
+     *
+     * @return the type of the macro content.
+     * @since 10.9-rc-1
+     */
+    @Unstable
+    default Type getType()
+    {
+        return new DefaultParameterizedType(null, List.class, Block.class);
```

Comment:
The default type should be ``String.class`` I think.

Reply:
Yes since previous behavior correspond to String type the default should be String.